docs/buildctl.md

```diff
@@ -0,0 +1,146 @@
+# buildctl
+
+`buildctl` is the command-line interface to `buildkitd`.
+
+```
+NAME:
+   buildctl - build utility
+
+USAGE:
+   buildctl [global options] command [command options] [arguments...]
+
+VERSION:
+   0.0.0+unknown
+
+COMMANDS:
+   du        disk usage
+   prune     clean up build cache
+   build, b  build
+   debug     debug utilities
+   help, h   Shows a list of commands or help for one command
+
+GLOBAL OPTIONS:
+   --debug                enable debug output in logs
+   --addr value           buildkitd address (default: "unix:///run/buildkit/buildkitd.sock")
+   --tlsservername value  buildkitd server name for certificate validation
+   --tlscacert value      CA certificate for validation
+   --tlscert value        client certificate
+   --tlskey value         client key
+   --tlsdir value         directory containing CA certificate, client certificate, and client key
+   --timeout value        timeout backend connection after value seconds (default: 5)
+   --help, -h             show help
+   --version, -v          print the version
+```
+
+## build
+
+Synopsis:
+
+```
+buildctl build --frontend dockerfile.v0 --opt target=foo --opt build-arg:foo=bar --local context=. --local dockerfile=. --output type=image,name=docker.io/username/image,push=true
+```
+
+`buildctl build` uses a buildkit daemon `buildkitd` to drive a build.
+
+The build consists of the following key elements:
+
+* frontend definition: parses the build descriptor, e.g. dockerfile
+* sources: sets relevant directories and files passed to the build
+* frontend options: options that are relevant to the particular frontend
+* output: defines what format of output to use and where to place it
+
+### frontend
+
+The frontend is declared by the flag `--frontend <frontend>`. The `<frontend>` must be one built into `buildkitd`, or an OCI
+image that implements the frontend API.
+
+In the above example, we are using the built-in `dockerfile.v0` frontend, which knows how to parse a dockerfile and convert it to LLB.
```

Comment:
We should call out both `dockerfile.v0` and `gateway.v0` if possible - `dockerfile.v0` is the one that's builtin, while `gateway.v0` is the one that allows connecting to "an oci image that implements the frontend API".

Reply:
See https://github.com/moby/buildkit/blob/master/README.md#exploring-dockerfiles.

Reply:
Updated.